kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/orchestration/ContainerOrchestrationService.java

```diff
@@ -41,7 +41,37 @@ public interface ContainerOrchestrationService {
      * @return a list of {@link ContainerInstanceDescriptor}s representing the available containers
      */
     public List<ContainerInstanceDescriptor> listContainerDescriptors();
-
+    
+    /**
+     * @since 2.4
+     * Lists all available images, returning the corresponding {@link ImageInstanceDescriptor}s
+     *
+     * @return a list of {@link ContainerInstanceDescriptor}s representing the available containers
+     */
+    public List<ImageInstanceDescriptor> listImageInstanceDescriptor();
+    
+    /**
+     * @since 2.4
+     * Deletes Image from container engine. Takes ImageID String as parameter. Will throw an error if image is being used by a container.
+     *
+     * @return void
+     * @throws KuraException 
+     */
+    public void deleteImage(String imageId) throws KuraException;
```

Comment:
/**
     * Deletes the specified image from the container engine. Takes ImageID String as parameter. Will throw an error if
     * image is being used by a container.
     *
     * @param imageId
     *            string parameter that identifies the image to be deleted
     * @throws KuraException
     *             if the image is used by a container or if the image deletion process fails
     * `@since 2.4`
     */